cmd/ethrex/cli.rs

```
@@ -367,62 +367,74 @@ pub async fn import_blocks(
+        let mut entries: Vec<_> = read_dir(path)
+            .expect("Failed to read blocks directory")
+            .map(|res| res.expect("Failed to open file in directory").path())
+            .collect();
+
+        // Sort entries to process files in order (e.g., 1.rlp, 2.rlp, ...)
+        entries.sort();
+
+        entries
+            .iter()
+            .map(|entry| {
+                let path_str = entry.to_str().expect("Couldn't convert path to string");
+                info!("Importing blocks from chain file: {path_str}");
+                utils::read_chain_file(path_str)
+            })
+            .collect()
     } else {
         info!("Importing blocks from chain file: {path}");
-        utils::read_chain_file(path)
+        vec![utils::read_chain_file(path)]
     };
-    let size = blocks.len();
-    for block in &blocks {
-        let hash = block.hash();
-        let number = block.header.number;
-        info!("Adding block {number} with hash {hash:#x}.");
-        // Check if the block is already in the blockchain, if it is do nothing, if not add it
-        let block_number = store.get_block_number(hash).await.map_err(|_e| {
-            ChainError::Custom(String::from(
-                "Couldn't check if block is already in the blockchain",
-            ))
-        })?;
 
-        if block_number.is_some() {
-            info!("Block {} is already in the blockchain", block.hash());
-            continue;
+    for blocks in chains {
+        let size = blocks.len();
+        // Execute block by block
+        for block in &blocks {
+            let hash = block.hash();
+            let number = block.header.number;
+            info!("Adding block {number} with hash {hash:#x}.");
+            // Check if the block is already in the blockchain, if it is do nothing, if not add it
+            let block_number = store.get_block_number(hash).await.map_err(|_e| {
+                ChainError::Custom(String::from(
```

nit: we could just ignore this error and just try to add the block